db/db_impl.cc

```diff
@@ -645,6 +668,7 @@ Status DBImpl::SyncWAL() {
     need_log_dir_sync = !log_dir_synced_;
   }
 
+  TEST_SYNC_POINT("DBWALTest::SyncWALNotWaitWrite:1");
```

Comment:
Sure. Will do. 
Regarding the current sync guarantees, I believe the idea was that push_back() could not be called while back() is accessed (otherwise back() would be stale). pop_front() could be called but according to our documentations it should be safe to call pop_front() and back() concurrently on deque. If it is not, I will update our documentation.